hack/test-e2e.sh

```diff
@@ -0,0 +1,116 @@
+#!/bin/bash
+set -euo pipefail
+
```

Comment:
Suggest to add a comment to explain prerequisites to running this test, like running through the guide first?